salt/utils/verify.py

```diff
@@ -287,29 +288,21 @@ def verify_env(
                 if "{}jobs".format(os.path.sep) in fsubdir:
                     continue
                 for root, dirs, files in salt.utils.path.os_walk(fsubdir):
-                    for name in files:
+                    for name in itertools.chain(files, dirs):
                         if name.startswith("."):
                             continue
                         path = os.path.join(root, name)
                         try:
                             fmode = os.stat(path)
                         except OSError:
-                            pass
-                        if fmode.st_uid != uid or fmode.st_gid != gid:
-                            if permissive and fmode.st_gid in groups:
-                                pass
-                            else:
-                                # chown the file for the new user
-                                os.chown(path, uid, gid)
-                    for name in dirs:
-                        path = os.path.join(root, name)
-                        fmode = os.stat(path)
+                            continue
                         if fmode.st_uid != uid or fmode.st_gid != gid:
                             if permissive and fmode.st_gid in groups:
                                 pass
                             else:
                                 # chown the file for the new user
                                 os.chown(path, uid, gid)
+
```

Comment:
**question** is it possible for this to try and chown a non-existent file?

Reply:
(given the race condition that appears to hit stat in the first place)

Reply:
ha, good catch, it's definitely possible that `os.chown` will raise exception while `os.stat` will still work (given we're talking about race conditions here)... I will update the PR